pairing/bn256/point.go

```diff
@@ -435,6 +435,64 @@ func (p *pointG2) String() string {
 	return "bn256.G2:" + p.g.String()
 }
 
+func (p *pointG2) Hash(m []byte) kyber.Point {
+	// Hash the data to get the "real" and "imaginary" parts of the
```

Comment:
Is this an existing standard that you are implementing, in which case, please mention it. Or are you syncing up to an existing implementation someplace else, in which case please reference it and add test vectors to the unit test that come from the other implementation.

Or is this your own invention?

It all looks reasonable to me, just trying to understand the context a bit more.

Reply:
There is no other implementation or standard for hashing to G2 on this curve (that we are aware of). As above, the choice of algorithm follows closely the existing hash-to-G1, but the choice of exactly how to hash the input data to the initial x coordinate is somewhat abritrary. That's why the unit tests that check explicit output values are marked as regression tests: they're to test future implementations against this implementation, not this implementation against some external implementation or standard.